examples/api-routes-graphql/pages/index.js

```diff
@@ -1,27 +1,29 @@
-import fetch from 'isomorphic-unfetch'
+import useSWR from 'swr'
 
-const Index = ({ users }) => (
-  <div>
-    {users.map((user, i) => (
-      <div key={i}>{user.name}</div>
-    ))}
-  </div>
-)
-
-Index.getInitialProps = async () => {
-  const response = await fetch('http://localhost:3000/api/graphql', {
+const fetcher = query =>
+  fetch('/api/graphql', {
     method: 'POST',
     headers: {
       'Content-type': 'application/json',
     },
-    body: JSON.stringify({ query: '{ users { name } }' }),
+    body: JSON.stringify({ query }),
   })
+    .then(res => res.json())
```

Comment:
Shouldn't this deal with HTTP errors?
```suggestion
    .then(res => res.ok ? res.json() : Promise.reject(new Error(`HTTP error ${res.status}`)))
```

Reply:
Check my answer [here](https://github.com/zeit/next.js/pull/11385#discussion_r399515769)